From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CoreWeave <>
Date: Wed, 11 Mar 2026 12:33:23 -0700
Subject: [PATCH] Fix FA2 + FA4 co-existence (#2331)

(cherry picked from commit 4219765dfdd8913bfe26134f748dd5ffcedd3c39)

Backported onto v2.8.3.post2, keeping __version__ at "2.8.3.post2".
---
 flash_attn/__init__.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/flash_attn/__init__.py b/flash_attn/__init__.py
index a7ad9f1e..91272aed 100644
--- a/flash_attn/__init__.py
+++ b/flash_attn/__init__.py
@@ -1,3 +1,8 @@
+from pkgutil import extend_path
+
+# look for every subdir with flash_attn base name such that fa2 and fa4 can be co-installed
+__path__ = extend_path(__path__, __name__)
+
 __version__ = "2.8.3.post2"
 
 from flash_attn.flash_attn_interface import (
-- 
2.34.1

